lib/SPIRV/SPIRVToOCL20.cpp

```diff
@@ -382,7 +382,11 @@ void SPIRVToOCL20::visitCallSPIRVAtomicBuiltin(CallInst *CI, Op OC) {
                                     .getFirstInsertionPt()));
           PExpected->setAlignment(CI->getType()->getScalarSizeInBits() / 8);
           new StoreInst(Args[1], PExpected, PInsertBefore);
-          Args[1] = PExpected;
+          unsigned AddrSpc = SPIRAS_Generic;
```

Comment:
What if use this address space directly in Alloca's constructor above?

Reply:
The LLVM IR Verifier rejects alloca's with an address space different from the DataLayout's alloca address space.
(so the assertion "Allocation instruction pointer not in the stack address space!" will fire if you do).